compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -232,6 +232,27 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 self.copy_clone_conditions(obligation)
             } else if Some(trait_def) == lang_items.clone_trait() {
                 self.copy_clone_conditions(obligation)
+            } else if Some(trait_def) == lang_items.as_repr_trait() {
+                let self_ty =
+                    self.infcx().shallow_resolve(obligation.predicate.skip_binder().self_ty());
+                match self_ty.kind() {
+                    ty::Adt(def, ..) => {
+                        if def.is_enum() {
+                            if def.repr.int.is_some() {
+                                BuiltinImplConditions::Where(Binder::dummy(vec![]))
```

Comment:
This means `AsRepr` is implemented for all enums, regardless whether it is fieldless or not.